features/minion/dominion/src/main/java/org/opennms/features/minion/dominion/MinionListener.java

```diff
@@ -0,0 +1,41 @@
+package org.opennms.features.minion.dominion;
+
+import java.util.Date;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.opennms.netmgt.dao.api.MinionDao;
+import org.opennms.netmgt.model.minion.OnmsMinion;
+
+public class MinionListener implements Processor {
+
+    private MinionDao minionDao;
+
+    public void process(Exchange exchange) throws Exception {
+
+        final String minionHandle = exchange.getIn().getBody(String.class);
```

Comment:
We should be passing a serialized version of MinionIdentity instead of a string here.